### ISSUE TITLE
Deprecation process

### ISSUE THREAD

**Issue by ethman** [interactiveaudiolab/nussl#147]
As nussl is changing and aging, we need a process to deprecate features that are obsolete, change names, or need to change. Right now, we have no such process despite a small community of users. We need to provide advance warnings so that users aren't blindsided by changes in the API, especially with the amount of churn that comes with a library that's in alpha like this is.

**Comment by pseeth** [comment]
Yep, we need a process for this still for sure. Let's discuss here when we get a chance to think about it!
